vm/systemSmartContracts/validator.go

```diff
@@ -1176,13 +1183,19 @@ func (v *validatorSC) unStakeTokens(args *vmcommon.ContractCallInput) vmcommon.R
 	return vmcommon.Ok
 }
 
-func (v *validatorSC) unStakeValue(registrationData *ValidatorDataV2, unStakeValue *big.Int) vmcommon.ReturnCode {
-	registrationData.TotalStakeValue.Sub(registrationData.TotalStakeValue, unStakeValue)
-	if registrationData.TotalStakeValue.Cmp(zero) < 0 {
-		v.eei.AddReturnMessage("contract error on unStakeTokens function, total stake < unstake value")
+func (v *validatorSC) processUnStakeValue(registrationData *ValidatorDataV2, unStakeValue *big.Int) vmcommon.ReturnCode {
+	unstakeValueIsOk := unStakeValue.Cmp(v.minUnstakeTokensValue) >= 0 || unStakeValue.Cmp(registrationData.TotalStakeValue) == 0
+	if !unstakeValueIsOk {
+		v.eei.AddReturnMessage("can not unstake the provided value either because is under the minimum threshold or " +
```

Comment:
I think this will break the backwards compatibility if it can happen in a situation that can now happen on the mainnet.
Strongly suggest as to run the new code if the topupV2 flag is set, otherwise run the exact same old code. We should have different .go files containing old/legacy code in order to keep the logic in hand.

Reply:
This is exactly the same as before. This does not have any backward compatibility issues.

Reply:
This means a lot of copy pasted code.

Reply:
unStakeValue can be called only if flag topup is enabled.